updated for 1.0.9 but untested yet!

this was for a nftables version from maybe 27 dec 2017 or earlier! (unknown version currently)

also show rule number when
$ sudo nft -a list ruleset

#diff --git a/src/rule.c b/src/rule.c
#index 7114380..66c23a0 100644
#--- a/src/rule.c
#+++ b/src/rule.c
#@@ -449,9 +449,12 @@ static void chain_print(const struct chain *chain)
# 		       hooknum2str(chain->handle.family, chain->hooknum),
# 		       chain->priority, chain_policy2str(chain->policy));
# 	}
#+  unsigned int c=0;
# 	list_for_each_entry(rule, &chain->rules, list) {
# 		printf("\t\t");
# 		rule_print(rule);
#+	  if (handle_output > 0)
#+      printf(" rule:%u", ++c); //helps identify rule from TRACE: outputs in logs (ie. when using: meta nftrace set 1)
# 		if (rule->handle.comment)
# 			printf(" comment \"%s\"\n", rule->handle.comment);
# 		else
Index: /var/tmp/portage/net-firewall/nftables-1.0.9/work/nftables-1.0.9/src/rule.c
===================================================================
--- .orig/var/tmp/portage/net-firewall/nftables-1.0.9/work/nftables-1.0.9/src/rule.c
+++ nftables-1.0.9/src/rule.c
@@ -492,7 +492,6 @@ void rule_print(const struct rule *rule,
 		if (!list_is_last(&stmt->list, &rule->stmts))
 			nft_print(octx, " ");
 	}
-
 	if (rule->comment)
 		nft_print(octx, " comment \"%s\"", rule->comment);
 
@@ -1068,8 +1067,10 @@ void chain_rules_print(const struct chai
 	if (chain->flags & CHAIN_F_BINDING)
 		octx->flags &= ~NFT_CTX_OUTPUT_HANDLE;
 
+	unsigned int c=0;
 	list_for_each_entry(rule, &chain->rules, list) {
-		nft_print(octx, "\t\t%s", indent ? : "");
+		nft_print(octx, "\t\trule:%u%s", ++c,indent ? : "");
+		//rule:number => helps identify rule from TRACE: outputs in logs (ie. when using: meta nftrace set 1)
 		rule_print(rule, octx);
 		nft_print(octx, "\n");
 	}
@@ -2797,3 +2798,4 @@ struct error_record *rule_postprocess(st
 	stmt_reduce(rule);
 	return NULL;
 }
+// vim: set tabstop=2 shiftwidth=2 noexpandtab :
